whelk-core/src/main/groovy/whelk/search2/Disambiguate.java

```diff
@@ -95,379 +108,91 @@ public Set<String> getAmbiguousEnumMapping(String alias) {
-                        }
-                );
+    
+    public Node expandChainAxiom(Property property) {
+        return _expandChainAxiom(property);
     }
 
-    private void addMappings(Map<?, ?> fromTermData, String toTermKey, TermType termType) {
-        String fromTermId = (String) fromTermData.get(JsonLd.ID_KEY);
-
-        addMapping(fromTermId, toTermKey, termType);
-        addMapping(toPrefixed(fromTermId), toTermKey, termType);
-
-        for (String prop : notatingProps) {
-            if (fromTermData.containsKey(prop)) {
-                addMapping((String) fromTermData.get(prop), toTermKey, termType);
-            }
-
-            String alias = (String) jsonLd.langContainerAlias.get(prop);
-
-            if (fromTermData.containsKey(alias)) {
-                Map<?, ?> byLang = (Map<?, ?>) fromTermData.get(alias);
-                for (String lang : jsonLd.locales) {
-                    List<?> values = JsonLd.asList(byLang.get(lang));
-                    values.forEach(v -> addMapping((String) v, toTermKey, termType));
-                }
-            }
-        }
+    private String getQueryCode(String property) {
+        return (String) getDefinition(property).get("librisQueryCode");
     }
 
-    private void addMapping(String from, String to, TermType termType) {
-        from = from.toLowerCase();
-
-        Map<String, String> aliasMappings = switch (termType) {
-            case CLASS -> classAliasMappings;
-            case PROPERTY -> propertyAliasMappings;
-            case ENUM -> enumAliasMappings;
-        };
-        Map<String, Set<String>> ambiguousAliases = switch (termType) {
-            case CLASS -> ambiguousClassAliases;
-            case PROPERTY -> ambiguousPropertyAliases;
-            case ENUM -> ambiguousEnumAliases;
-        };
-
-        if (ambiguousAliases.containsKey(from)) {
-            ambiguousAliases.get(from).add(to);
-        } else if (aliasMappings.containsKey(from)) {
-            if (aliasMappings.get(from).equals(to)) {
-                return;
-            }
-            ambiguousAliases.put(from, new HashSet<>(Arrays.asList(to, aliasMappings.remove(from))));
-        } else {
-            aliasMappings.put(from, to);
+    private Optional<Map<?, ?>> getDefinition(Map<?, ?> link, Whelk whelk) {
+        var iri = get(link, ID_KEY, "");
+        if (iri.isEmpty()) {
+            return Optional.empty();
         }
+        var fromVocab = get(vocab, jsonLd.toTermKey(iri), Map.of());
+        return fromVocab.isEmpty() ? loadThing(iri, whelk) : Optional.of(fromVocab);
     }
 
     private static boolean isKbvTerm(Map<?, ?> termDefinition) {
-        return getAsOptionalMap(termDefinition, "isDefinedBy")
-                .flatMap(Disambiguate::getLinkIri)
-                .filter("https://id.kb.se/vocab/"::equals)
-                .isPresent();
+        return "https://id.kb.se/vocab/".equals(get(termDefinition, List.of(Rdfs.IS_DEFINED_BY, ID_KEY), ""));
```

Comment:
Get from `@vocab` in system context instead?
("isSystemVocabTerm")

Reply:
Good idea